### ISSUE TITLE
New Dashboards

### ISSUE THREAD

**Issue by niemyjski** [exceptionless/Exceptionless#229]
WIP
- [ ] Break down of event types, 
- [ ] number of users online, 
- [ ] percent of users getting an error, 
- [ ] where are the people located geographically, 
- [ ] what browsers are they using, 

- [ ] New vs. returning users (pie chart?), 
- [ ] active users and sessions graph, 
- [ ] geo (top country usage), 
- [ ] top language, 
- [ ] top app versions, 
- [ ] top devices or resolutions, 
- [ ] billing data (how long until my next billing reset and how far into data cap i am)

- [ ] number of sessions, 
- [ ] number of users, 
- [ ] top stacks, 
- [ ] top feature usage, 
- [ ] recent errors, 
- [ ] avg session duration number, 
- [ ] session/hour number, 
- [ ] how long are their sessions...

**Comment by BillHenning** [comment]
A lot of other systems split out some of what you have in "Update existing Sessions dash" into their own pages.  I'm fine with having it be most consolidated onto one though.  The existing Recent Sessions list on that page should fall under all those other proposed new items.

**Comment by niemyjski** [comment]
I agree, I just don't know where you'd put it and if a user is already looking a ton of user metrics, it makes sense to put it there.

**Comment by niemyjski** [comment]
Maybe we create a new default dashboard that could have a lot of this like:
- billing (events left and when it renews)
- top projects
- error stacks
- current user count last 24 hours with count of currently active users
- users geo
- % of those that had an error
- pie chart with active versions

each of those charts could be clickable and take you to a different dashboard that shows you more info

**Comment by BillHenning** [comment]
I would like to have an overview dashboard like that and have it be your default landing page.

**Comment by niemyjski** [comment]
Yeah, that's what it would be. I think we would have to come up with a new name or rename the existing /dashboard route

**Comment by niemyjski** [comment]
Feature request: simple pie chart of Feature Most Frequent" to show proportions.

**Comment by niemyjski** [comment]
Being able to see all events fixed in a specific version.

**Comment by john-roland** [comment]
Hello Blake,

some few requests regarding dashboards..

Feature request: have the possibility to add an new entry in the left side menu for our own event type, or that exceptionless automatically detect it. in my case it's a "metric" type.

Feature request: on stack dashboard, display a pie chart of tags repartition on the events.

Feature request: on project & stack dashboard, display tags as filtering link, so a click on it will filters the events on the actual tag.

**Comment by niemyjski** [comment]
Those are solid suggestions and I can see us doing those :)

**Comment by niemyjski** [comment]
If there is anything else you can think of let us know. We need all the help and feedback we can get! P.S., It's on our road map too :) https://exceptionless.com/2017-exceptionless-feature-functionality-and-enhancement-roadmap/

**Comment by john-roland** [comment]
ok, I will... actually I have a lot of ideas and needs  ;)
I now use ExceptionLess at a very large scale (more than 1 million events / week)..

Tell me how is the best way to submit this to you

**Comment by niemyjski** [comment]
GitHub issues is the best way so we can track and have a discussion. we are also on slack (http://slack.exceptionless.com)! I'd be more than happy to meet up with you as well and talk every your needs and pain points.

P.S., how's the last 4 days been in terms of dashboard performance :)?

**Comment by john-roland** [comment]
ok, i will talk with my collegues to collect their feedback and post something here

It would be great to meet you, but if you remember me from .nettiers, I'm in France, oups Geneva (switzerland) now :)

**Comment by niemyjski** [comment]
Yeah, I remember the legendary John Roland :). We could figure out a time that works (I meet with people all over the world :)). Just send an in app message or message on slack and we'll setup a time!

**Comment by davidkeaveny** [comment]
Custom dashboards would the the next killer feature for me (after environments). I want to be able to report on the number of exceptions logged over time by application (being able to break that down into new and unique is also good), so that both management and the development team can see the overall trend, whether bug killing sessions are delivering measurable results, whether our codebase is getting better or worse over time. 

With our old system, I had to trawl through literally thousands of notification emails in a mailbox, scanning the exception reports to generate a final report (I automated it all, of course, but it was still a lengthy process). Having it all right there in a dashboard would be beyond awesome.

**Comment by niemyjski** [comment]
We are starting on #177, we feel this is needed first

**Comment by nmg196** [comment]
Wouldn't it make more sense to integrate with Google Analytics for this stuff, instead of reinventing the wheel? I'm guessing nearly all sites running ExceptionLess are either already running GA as well, or easily could with 6 lines of code :)

**Comment by niemyjski** [comment]
@nmg196 yes and no, we have the data for a lot more charts and the backend supports dynamic stats, we just need help building out the front entd.

**Comment by niemyjski** [comment]
We are making some big updates to how our dashboards are going to work in the upcoming version but haven't forgotten about this issue.

**Comment by ejsmith** [comment]
Closing this as part of an issue cleanup.

This is an older dashboard planning checklist and several items have either been implemented elsewhere, changed shape, or no longer match the current Svelte UI/dashboard direction. Keeping this broad umbrella open makes the backlog harder to act on.

Please open focused issues for any specific dashboard metric or visualization still needed in the latest app.
